Framework/DataObjects/src/EventList.cpp

```diff
@@ -3753,6 +3771,46 @@ void EventList::filterByTimeAtSample(Types::Core::DateAndTime start, Types::Core
   }
 }
 
+/** Filter this EventList into an output EventList, using TimeROI
+ * keeping only events within the >= start and < end pulse times.
```

Comment:
Since a `TimeROI` object may contain more than one time interval, the function is keeping events with pulse times within any of the ROI time intervals and discarding events within any of the masked time intervals.  Can you insert this or a similar sentence in the docstring?

Reply:
The comment has been updated to better reflect the actions performed in the function.